rs/bitcoin/kyt/src/fetch/tests.rs

```diff
@@ -37,22 +37,19 @@ impl FetchEnv for MockEnv {
     }
 
     fn config(&self) -> Config {
-        Config {
-            btc_network: BtcNetwork::Mainnet,
-            kyt_mode: KytMode::Normal,
-        }
+        Config::new_and_validate(BtcNetwork::Mainnet, KytMode::Normal).unwrap()
     }
 
     async fn http_get_tx(
         &self,
-        provider: Provider,
+        provider: &Provider,
         txid: Txid,
         max_response_bytes: u32,
     ) -> Result<Transaction, HttpGetTxError> {
         self.calls
             .borrow_mut()
             .push_back((txid, max_response_bytes));
-        *self.called_provider.borrow_mut() = Some(provider);
+        *self.called_provider.borrow_mut() = Some(provider.clone());
```

Comment:
I agree. What's the reason for the proposed change?

Reply:
Both `BtcNetwork` and `Provider` used be `Copy`, but since `BtcNetwork::Regtest` now takes a string argument, they cannot be `Copy` any more. Here it is implementing a class method, it may see awkward. But using a reference makes more sense elsewhere.